### ISSUE TITLE
[Bug]: cn upgrader bug

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#12262]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 1.0-dev
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

| internal error: unknown d | internal error: unknown data type collati | internal error: unknown data type collati
(1)
Wraps: (2)
  -- stack trace:
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.ParseDataTypeToColType
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader_util.go:56
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).GetCurrentSchema
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:69
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).UpgradeNewViewColumn
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:208
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).Upgrade
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:171
  | github.com/matrixorigin/matrixone/pkg/cnservice.(*service).upgrade
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/server_task.go:130
  | github.com/matrixorigin/matrixone/pkg/cnservice.(*service).handleCommands
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/server_heartbeat.go:97
  | github.com/matrixorigin/matrixone/pkg/cnservice.(*service).heartbeat
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/server_heartbeat.go:85
  | github.com/matrixorigin/matrixone/pkg/cnservice.(*service).heartbeatTask
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/server_heartbeat.go:52
  | github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1
  |     /go/src/github.com/matrixorigin/matrixone/pkg/common/stopper/stopper.go:259
  | runtime.goexit
  |     /usr/local/go/src/runtime/asm_amd64.s:1598
Wraps: (3) internal error: unknown data type collati
Error types: (1) *errutil.withContext (2) *errutil.withStack (3) *moerr.Error |
| internal error: unknown d | internal error: unknown data type created | internal error: unknown data type created
(1)
Wraps: (2)
  -- stack trace:
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.ParseDataTypeToColType
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader_util.go:56
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).GetCurrentSchema
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:69
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).UpgradeNewViewColumn
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:208
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).Upgrade
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:171

**Comment by fengttt** [comment]
And this,

| internal error: can not g | internal error: can not get the schema | internal error: can not get the schema
(1)
Wraps: (2)
  -- stack trace:
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).GetCurrentSchema
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:79
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).UpgradeNewViewColumn
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:208
  | github.com/matrixorigin/matrixone/pkg/cnservice/upgrader.(*Upgrader).Upgrade
  |     /go/src/github.com/matrixorigin/matrixone/pkg/cnservice/upgrader/upgrader.go:171
  | github.com/matrixorigin/matrixone/pkg/cnservice.(*service).upgrade

**Comment by gavinyue** [comment]
Looking

**Comment by gavinyue** [comment]
正在看 已经找到问题

**Comment by gavinyue** [comment]
The problem is that the upgrader only support limited column types now.  
Will add current required.

**Comment by gavinyue** [comment]
{"level":"ERROR","time":"2023/10/26 01:27:44.663221 +0000","caller":"upgrader/upgrader.go:193","msg":"error: internal error: Upgrade failed during system startup! The upgrade error message is listed below: \nerror[0]: CN upgrade table or view 'information_schema.processlist' under tenant 'sys:0' reports error: internal error: unknown data type host\u0000\u0000\u0000\ninternal error: unknown data type node_ty\ninternal error: unknown data type node_uu\ninternal error: unknown data type query_type\u0000\u0000\ninternal error: unknown data type request\ninternal error: unknown data type respons\ninternal error: unknown data type result_\ninternal error: unknown data type role_id\ninternal error: unknown data type rows_re\ninternal error: unknown data type session\ninternal error: unknown data type sql_sou\ninternal error: unknown data type stateme\ninternal error: unknown data type stateme\ninternal error: unknown data type stateme\ninternal error: unknown data type stateme\ninternal error: unknown data type stats\u0000\u0000\ninternal error: unknown data type status\u0000\ninternal error: unknown data type transac\ninternal error: unknown data type user\u0000\u0000\u0000\ninternal error: unknown data type __mo_fa\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type account\u0000\u0000\u0000\u0000\u0000\ninternal error: unknown data type collect\ninternal error: unknown data type metric_\ninternal error: unknown data type node\u0000\u0000\u0000\ninternal error: unknown data type role\u0000\u0000\u0000\ninternal error: unknown data type type\u0000\u0000\u0000\ninternal error: unknown data type value\u0000\u0000\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type account\ninternal error: unknown data type collect\ninternal error: unknown data type node\u0000\u0000\u0000\ninternal error: unknown data type role\u0000\u0000\u0000\ninternal error: unknown data type type\u0000\u0000\u0000\ninternal error: unknown data type value\u0000\u0000\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type account\ninternal error: unknown data type collect\ninternal error: unknown data type node\u0000\u0000\u0000\ninternal error: unknown data type role\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\ninternal error: unknown data type type\u0000\u0000\u0000\ninternal error: unknown data type value\u0000\u0000\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type account\ninternal error: unknown data type collect\ninternal error: unknown data type node\u0000\u0000\u0000\ninternal error: unknown data type role\u0000\u0000\u0000\ninternal error: unknown data type value\u0000\u0000\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type account\ninternal error: unknown data type collect\ninternal error: unknown data type node\u0000\u0000\u0000\ninternal error: unknown data type role\u0000\u0000\u0000\ninternal error: unknown data type type\u0000\u0000\u0000\ninternal error: unknown data type value\u0000\u0000\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type account\ninternal error: unknown data type collecttime\u0000\ninternal error: unknown data type node\u0000\u0000\u0000\ninternal error: unknown data type role\u0000\u0000\u0000\ninternal error: unknown data type value\u0000\u0000\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type account\ninternal error: unknown data type collect\ninternal error: unknown data type node\u0000\u0000\u0000\ninternal error: unknown data type role\u0000\u0000\u0000\ninternal error: unknown data type value\u0000\u0000\ninternal error: unknown data type __mo_fa\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type column_\ninternal error: unknown data type constra\ninternal error: unknown data type constra\ninternal error: unknown data type ordinal\ninternal error: unknown data type table_c\ninternal error: unknown data type table_n\ninternal error: unknown data type table_schema\ninternal error: unknown data type __mo_fa\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type block_o\ninternal error: unknown data type block_o\ninternal error: unknown data type context\ninternal error: unknown data type cpu_sys\ninternal error: unknown data type cpu_use\ninternal error: unknown data type duratio\ninternal error: unknown data type message\ninternal error: unknown data type message\ninternal error: unknown data type page_fa\ninternal error: unknown data type page_fa\ninternal error: unknown data type query_i\ninternal error: unknown data type seq\u0000\u0000\u0000\u0000\ninternal error: unknown data type source_\ninternal error: unknown data type source_\ninternal error: unknown data type source_\ninternal error: unknown data type state\u0000\u0000\u0000\u0000\u0000\u0000\u0000\ninternal error: unknown data type swaps\u0000\u0000\ninternal error: unknown data type __mo_ro\ninternal error: unknown data type account\ninternal error: unknown data type client_\ninternal error: unknown data type command\ninternal error: unknown data type conn_id\ninternal error: unknown data type db\u0000\u0000\u0000\u0000\u0000\ninternal error: can not get the schema "}
--


<br class="Apple-interchange-newline">

**Comment by gavinyue** [comment]
A similar error in loki. fix them together

**Comment by gavinyue** [comment]
https://github.com/matrixorigin/matrixone/pull/12409

one pr to fix the found problem for mo_cache upgrade

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Blocked on another pr

**Comment by gavinyue** [comment]
https://github.com/matrixorigin/matrixone/pull/12409

需要这个PR

**Comment by gavinyue** [comment]
但是为什么column name 和type 弄错的问题，  还没复现， 这个只能留到1.2了

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Pending

**Comment by arjunsk** [comment]
This issue has re-occurred in MO Cloud FreeTier01 (running 1.1-dev)

![image](https://github.com/matrixorigin/matrixone/assets/9638314/51d1043c-dc2a-4e2e-8608-5fd5695b53a6)

Graphana Link: [here](http://47.96.110.240/explore?panes=%7B%2219h%22:%7B%22datasource%22:%22loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bmatrixone_cloud_main_cluster%3D%5C%22freetier-01%5C%22,%20namespace%3D%5C%22freetier-01%5C%22%7D%20%7C%3D%20%60upgrade%60%20%7C%3D%20%60mo_catalog%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-7d%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1&orgId=1)

**Comment by arjunsk** [comment]
I wanted to check, why are we ignoring the error returned by ug.Upgrade(moServerCtx) [here](https://github.com/arjunsk/matrixone/blob/179582e8510c1f33977f5992ac0c0e36311d108f/pkg/cnservice/server_task.go#L138) 

Incase of an upgrade failure we should panic and let the user know that the upgrade is not successful.

**Comment by gavinyue** [comment]
等待处理

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Pending

**Comment by gavinyue** [comment]
Pending

**Comment by daviszhen** [comment]
未投入

**Comment by qingxinhome** [comment]
The fix for this bug depends on the refactoring of the framework upgrade #12262

**Comment by qingxinhome** [comment]
fixing

**Comment by qingxinhome** [comment]
PR has been submitted

**Comment by qingxinhome** [comment]
PR has been submitted

**Comment by qingxinhome** [comment]
PR has been submitted

**Comment by qingxinhome** [comment]
处理中

**Comment by qingxinhome** [comment]
处理中

**Comment by qingxinhome** [comment]
Pending processing

**Comment by qingxinhome** [comment]
待合并

**Comment by qingxinhome** [comment]
The issue has been resolved and PR has been submitted. Please verify and test it @fengttt @sukki37 @aressu1985 @daviszhen
